openmrs/omod/src/main/java/org/projectbuendia/openmrs/web/controller/ProfileManager.java

```diff
@@ -145,68 +175,79 @@ String getNextVersionedFilename(String name) {
     }
 
     /** Handles an uploaded profile. */
-    // TODO: Give some error message if the profile is not uploaded correctly
-    //       (When tomcat7 didn't have permission to PROFILES_DIR it was failing
-    //        without showing any error message to the user)
-    void addProfile(MultipartHttpServletRequest request, ModelMap model) {
+    private void addProfile(MultipartHttpServletRequest request, ModelMap model) {
         List<String> lines = new ArrayList<>();
         MultipartFile mpf = request.getFile("file");
         if (mpf != null) {
+            String message = "";
+            String filename = mpf.getOriginalFilename();
             try {
                 File tempFile = File.createTempFile("profile", null);
                 mpf.transferTo(tempFile);
-                if (execute(VALIDATE_CMD, tempFile, lines)) {
-                    String filename = getNextVersionedFilename(mpf.getOriginalFilename());
-                    File newFile = new File(PROFILE_DIR, filename);
-                    model.addAttribute("filename", filename);
+                boolean exResult = execute(VALIDATE_CMD, tempFile, lines);
```

Comment:
I agree with @llvasconcellos. It makes the code more clear.
In my opinion, we should keep if clauses side-effect free.
The only function of an if statement is to test whether a condition is true.